src/array.cr

```diff
@@ -862,11 +862,30 @@ class Array(T)
     self
   end
 
+  # Removes all elements from `self` for which the given block returns falsy value.
+  # 
+  # if no elements found, self become an empty array.
```

Comment:
Ah I thought to drop these, it's implicit for me me from the above and also shown in the examples. But if we want to make it explicit I think a better wording would be "If all elements are removed the array becomes empty."